### DISCUSSION TITLE
Recommended method to edit portfile in manifest mode

### DISCUSSION THREAD

**Discussion by sidy3d** [microsoft/vcpkg#23113] [Q&A] [answered]
Hi,

I am starting to use manifest mode through a Visual Studio C++ project using the `vcpkg.json` file. I am using the PCL library and a few other libraries. 

Due to some custom configurations and avoiding compiler bugs I am directly editing the `portfile.cmake` file. This includes enabling the OpenMP feature for increased performance and disabling OpenNurbs feature due to some compile errors with MSVC v143  #19561. This is done by manually editing the lines containing `DBUILD_surface_on_nurbs` and `DWITH_OPENMP` in the in the `vcpkg\ports\pcl\portfile.cmake` file.

```
...
vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
    OPTIONS
        # BUILD
        -DBUILD_surface_on_nurbs=OFF
        # PCL
        -DPCL_BUILD_WITH_BOOST_DYNAMIC_LINKING_WIN32=${PCL_SHARED_LIBS}
        -DPCL_BUILD_WITH_FLANN_DYNAMIC_LINKING_WIN32=${PCL_SHARED_LIBS}
        -DPCL_BUILD_WITH_QHULL_DYNAMIC_LINKING_WIN32=${PCL_SHARED_LIBS}
        -DPCL_SHARED_LIBS=${PCL_SHARED_LIBS}
        # WITH
        -DWITH_PNG=ON
        -DWITH_QHULL=OFF
        -DWITH_OPENNI=OFF
        -DWITH_ENSENSO=OFF
        -DWITH_DAVIDSDK=OFF
        -DWITH_DSSDK=OFF
        -DWITH_RSSDK=OFF
        -DWITH_RSSDK2=OFF
        -DWITH_OPENMP=ON
        # FEATURES
        ${FEATURE_OPTIONS}
    MAYBE_UNUSED_VARIABLES
        PCL_BUILD_WITH_FLANN_DYNAMIC_LINKING_WIN32
)
...
```

Is this there a recommended way to perform this "patching"?

Thanks,

**Comment by autoantwort** [comment, marked as answer]
Yes, there is https://github.com/microsoft/vcpkg/blob/master/docs/examples/patching.md

**Reply by sidy3d** [reply]
This still requires storing a `.patch` file in the `vcpkg\ports\pcl\portfile` directory. Is there a way to tell the manifest file to include the `.patch` file from a different path?

**Reply by autoantwort** [reply]
You can use overlay ports
